app/controllers/server.rb

```diff
@@ -1,9 +1,108 @@
+require_relative '../models/client'
+require_relative '../models/processor'
+require 'pry'
+
 module RushHour
   class Server < Sinatra::Base
-    not_found do
+    def not_found(error_type)
+      @error_message = error_type
       erb :error
     end
+
+    get "/" do
+      erb :dashboard
+    end
     
-    # Write the rest of your controller code here!
+    get "/sample_sender" do
+      erb :sample_sender
+    end
+
+    get "/sources" do
+      status 200
+      erb :sources
+    end
+
+    get "/sources/" do
+      return (status 400) && not_found(page_not_found)
+      erb :error
+    end
+
+    get "/login" do
+      erb :login
+    end
+
+    get "/sources/:identifier" do |client_identifier|
+      return not_found("Client does not exist.") if Client.find_by(identifier: client_identifier).nil?
+      return not_found("No payloads for client.") if Client.find_by(identifier: client_identifier).payloads.empty?
+      @client = Client.find_by(identifier: client_identifier)
+      @events = @client.all_event_names
+      erb :sources_user
+    end
+
+    post "/sources" do
+      client_params = Hash.new
+      client_params[:identifier] = params["identifier"]
+      client_params[:root_url] = params["rootUrl"]
+      return status 400 unless Client.new(client_params).valid?
+      return status 403 if Client.pluck(:identifier).include?(client_params[:identifier])
+      Client.create(client_params)
+    end
+
+    post "/sources/:identifier/data" do |identifier|
```

Comment:
we could definitely extract all of this logic out to a different method - similar to the `post '/sources'`